data/scripts.js

```diff
@@ -2969,8 +2969,32 @@ exports.BattleScripts = {
 
 		// We choose level based on BST. Min level is 70, max level is 99. 600+ BST is 70, less than 300 is 99. Calculate with those values.
 		// Every 10.35 BST adds a level from 70 up to 99. Results are floored. Uses the Mega's stats if holding a Mega Stone
-		// To-do: adjust levels of mons with boosting items (Light Ball, Thick Club etc)
 		var bst = template.baseStats.hp + template.baseStats.atk + template.baseStats.def + template.baseStats.spa + template.baseStats.spd + template.baseStats.spe;
+		// Adjust levels of mons based on abilities (Pure Power, Sheer Force, etc.) and also Eviolite
+		// If a mon's offense stat is increased by M times that stat, with 255 EV, its effective base stat becomes (1+M) * ([base stat] + 50).
+		// If a mon's defensive stat is increased by M times that stat, with 0 EV, its effective base stat becomes (1+M) * ([base stat] + 18).
+		// Cut some ability cases a little slack since many of these mons can't have any other kind of boosting item (based on holistic judgment)
+		var templateAbility = (baseTemplate === template ? ability : template.abilities[0]);
+		if (templateAbility === 'Huge Power' || templateAbility === 'Pure Power') {
+			bst += template.baseStats.atk + 45;
```

Comment:
As mentioned in the comments, if you increase a stat by `M`, then it increases the effective base stat by `M * [normalized base stat] = M * ([base stat]+50)`. In this case, since Attack goes up by an equal amount, `M=1`. Using 45 instead of 50 is holistic judgment.